Bottom: 33bb85cbcce6261a0e381b612d28a1a8f0cbe64d
Top:    bb71773f72decca0c522eb1ea891e04541be7b0e
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-22 05:41:18 +0100

Try debugging draw speed




---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 7204be9..bb9999e 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -2165,13 +2165,12 @@ Benchmark (int argc, char **argv, int x, int y)
     {
       gdk_window_invalidate_rect (gport->drawing_area->window, NULL, 1);
       gdk_window_process_updates (gport->drawing_area->window, FALSE);
-      gdk_display_sync (display);
       time (&end);
       i++;
     }
   while (end - start < 10);
 
-  printf ("%g redraws per second\n", i / 10.0);
+  printf ("%g redraws per second\n", (double)i / (double)(end-start));
 
   return 0;
 }
